src/test/java/com/pubnub/api/managers/FastSubscriptionManagerTest.java

```diff
@@ -65,6 +67,30 @@ public void performsLongPollingAfterTimeout() throws IllegalAccessException {
         });
     }
 
+    @Test
+    public void reconnectRestartsTheLoopRegardlessOfState() {
+        AtomicInteger numberOfCalls = new AtomicInteger(0);
+        final ResponseSupplier<SubscribeEnvelope> responseSupplier = requestDetails -> {
+            numberOfCalls.incrementAndGet();
+            return new ResponseHolder<>(new SocketException(
+                    "socket exception"));
+        };
+
+        final RetrofitManager retrofitManagerMock = retrofitManagerMock(responseSupplier);
+
+        final SubscriptionManager subscriptionManager = subscriptionManagerUnderTest(retrofitManagerMock);
+
+        final SubscribeOperation subscribeOperation = SubscribeOperation.builder()
+                .channels(singletonList("ch1"))
+                .channelGroups(emptyList())
+                .build();
+
+        subscriptionManager.adaptSubscribeBuilder(subscribeOperation);
+
+        subscriptionManager.reconnect();
```

Comment:
To my knowledge, reconnect happens only when it is disconnected. Without disconnect, is it logical to reconnect?

Reply:
Yes, it's logical. User can call this regardless of the state

Reply:
what happens when reconnect is called multiple times?

Reply:
We will stop any currently ongoing call, and restart the loop

Reply:
ok. thanks for clarifying.

Reply:
No problem!